agent/engine/docker_container_engine.go

```diff
@@ -388,7 +397,14 @@ func (dg *dockerGoClient) createContainer(ctx context.Context, config *docker.Co
 	if err != nil {
 		return DockerContainerMetadata{Error: CannotCreateContainerError{err}}
 	}
-	return dg.containerMetadata(dockerContainer.ID)
+	//TODO DEBUG Uncomment return dg.containerMetadata(dockerContainer.ID)
+	md := dg.containerMetadata(dockerContainer.ID)
+	if (md.Metadata == nil) {
+//		os.Stderr.WriteString("EDTEST metadata is nil at createContainer\n") //TODO DEBUG
```

Comment:
I've removed most of these debug statements and replaced them with more appropriate seelog messages (Mostly for error handling and activity tracking).